src/Auth0.ManagementApi/ManagementApiClient.cs

```diff
@@ -206,7 +206,8 @@ protected virtual void Dispose(bool disposing)
         {
             if (!disposed && disposing)
             {
-                _apiConnection.Dispose();
+                if (connection is IDisposable disposableConnection)
+                    disposableConnection.Dispose();
```

Comment:
Given that an `IApiConnection` doesn't necessarily implement `IDisposable` like `ApiConnection` does we have to check it does first before we try to dispose.